pkg/util/fs_util.go

```diff
@@ -188,9 +188,12 @@ func extractFile(dest string, hdr *tar.Header, tr io.Reader) error {
 	switch hdr.Typeflag {
 	case tar.TypeReg:
 		logrus.Debugf("creating file %s", path)
-		// It's possible a file is in the tar before its directory.
-		if _, err := os.Stat(dir); os.IsNotExist(err) {
+		// It's possible a file is in the tar before its directory,
+		// or a file was copied over a directory prior to now
+		fi, err := os.Stat(dir)
+		if os.IsNotExist(err) || !fi.IsDir() {
 			logrus.Debugf("base %s for file %s does not exist. Creating.", base, path)
+			os.RemoveAll(dir)
```

Comment:
this is dangerous. Are we sure we want to do this?

Reply:
Agreed.  We should still try to do a os.MkdirAll here, but if that fails we should fail out entirely.